central/complianceoperator/v2/report/manager/format/formatter_test.go

```diff
@@ -204,3 +289,37 @@ func compareStringSlice(a []string, b []string) bool {
 func (m *valueMatcher) String() string {
 	return m.error
 }
+
+type failedClusterValueMatcher struct {
+	data   map[string][]*report.ResultRow
+	failed map[string]*storage.ComplianceOperatorReportSnapshotV2_FailedCluster
+	error  string
+}
+
+func (m *failedClusterValueMatcher) Matches(target interface{}) bool {
```

Comment:
I'm not sure if my understanding is correct - we are comparing if `m` is equal to `target` here?

If yes, I would suggest doing it differently (if it is possible). We could use: `gomock.Cond` with a callback function. And we can use `s.Equal` in the callback to compare values.

Reply:
I've changed the matchers (expect for the `emptyValueMatcher`) for `gomock.Cond`. Let me know what you think.